xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -371,8 +378,8 @@ static final class EdfScheduler {
     void add(int index, double weight) {
       checkArgument(weight > 0.0, "Weights need to be positive.");
       ObjectState state = new ObjectState(Math.max(weight, MINIMUM_WEIGHT), index);
-      state.deadline = 1 / state.weight;
-      // TODO(zivy): randomize the initial deadline.
+      // Randomize the initial deadline.
+      state.deadline = (1 - random.nextDouble()) * (1 / state.weight);
```

Comment:
Is there a reason for `1-randomDouble` instead of just `randomDouble`? I see that flips the range to be inclusive of 1 and exclusive of 0, but that seems to not matter here. Shouldn't matter either way, but seemed strange.